src/front/glsl/context.rs

```diff
@@ -1069,35 +1069,138 @@ impl Context {
                 )?;
                 return Ok((maybe_expr, meta));
             }
+            // `HirExprKind::Conditional` represents the ternary operator in glsl (`:?`)
+            //
+            // The ternary operator is defined to only evaluate one of the two possible
+            // expressions which means that it's behavior is that of an `if` statement,
+            // and it's merely syntatic sugar for it.
             HirExprKind::Conditional {
                 condition,
                 accept,
                 reject,
             } if ExprPos::Lhs != pos => {
```

Comment:
This match arm is 122 lines long. Would it make sense to move it into its own function or `Context` method?

Reply:
Yes this part needs to be refractored a bit but I thimk that should be done in another PR.